mm2src/coins/eth.rs

```diff
@@ -2609,10 +2631,21 @@ impl MarketCoinOps for EthCoin {
         let coin = self.clone();
 
         let fut = async move {
-            coin.block_number()
-                .await
-                .map(|res| res.as_u64())
-                .map_err(|e| ERRL!("{}", e))
+            if coin.is_tron() {
+                // TODO: Implement a TRON client
+                warn!(
+                    "Called EthCoin::current_block() for TRON; \
+                         returning stub value {} until TRON client is implemented",
+                    1u64
+                );
+                // Temporary TRON stub until full client support is available
+                Ok(1u64)
+            } else {
+                coin.block_number()
+                    .await
+                    .map(|res| res.as_u64())
+                    .map_err(|e| ERRL!("{}", e))
+            }
```

Comment:
nit: i think that's more self-contained
```rust
        let fut = async move {
            if coin.is_tron() {
                // TODO: Implement a TRON client
                warn!(
                    "Called EthCoin::current_block() for TRON; \
                         returning stub value {} until TRON client is implemented",
                    1u64
                );
                // Temporary TRON stub until full client support is available
                return Ok(1u64);
            }
            coin.block_number()
                .await
                .map(|res| res.as_u64())
                .map_err(|e| ERRL!("{}", e))
        };
```

Reply:
done https://github.com/KomodoPlatform/komodo-defi-framework/pull/2467/commits/1f69c9d0e62b5e8f996f29f04c50a51a2e6baec8